contracts/core/Dispatcher.sol

```diff
@@ -425,18 +477,17 @@ contract Dispatcher is OwnableUpgradeable, UUPSUpgradeable, IDispatcher {
         emit RecvPacket(receiver, packet.dest.channelId, packet.sequence);
 
         // If pkt is already timed out, then return early so dApps won't receive it.
-        if (_isPacketTimeout(packet)) {
+        if (_isPacketTimeout(packet.timeoutTimestamp, packet.timeoutHeight.revision_height)) {
```

Comment:
changes to the timeout behaviour should not be part of this commit. As a general rule, we should have features being added/fixed within individual commits in case we need to deploy/revert them individually. Mixing them will potentially cause unwanted side effects that will be very hard to predict.

Reply:
noted - these should be split into separate commits now, thanks for the suggestion @nicopernas !